index.js

```diff
@@ -9,6 +10,15 @@ module.exports = createXHR
 createXHR.XMLHttpRequest = window.XMLHttpRequest || noop
 createXHR.XDomainRequest = "withCredentials" in (new createXHR.XMLHttpRequest()) ? createXHR.XMLHttpRequest : window.XDomainRequest
 
+forEach(["get", "put", "post", "patch", "head", "delete"], function(method) {
```

Comment:
I read the content of for-each and we don't need 80% of it. For a simple hardcoded array of strings it can be much simpler.
I'll push it out in few minutes